lib/phoenix_live_view/utils.ex

```diff
@@ -347,6 +363,17 @@ defmodule Phoenix.LiveView.Utils do
   defp validate_mount_redirect!({:live, {_, _}, _}), do: raise_bad_mount_and_live_patch!()
   defp validate_mount_redirect!(_), do: :ok
 
+  @doc """
+  Calls the `handle_params/3` callback if it has been exported by the given `view`.
+  """
+  def maybe_call_handle_params(%Socket{} = socket, %{exported?: true} = lifecycle, params, uri) do
```

Comment:
Isn't this still only calling the hooks it if it is exported? :) When we get here, we already checked that `lifecycle.any?` returned true, so there is for sure something to call. I would likely remove this function and move the check to after `case Lifecycle.handle_params(params, uri, socket) do` a couple lines below. We should probably add an integration test to verify if the callbacks are indeed being called even without the function definition.

Reply:
Yep, you're totally right! 🙈

I will definitely add tests for that, too :)

Reply:
Ready for another look! :)